src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs

```diff
@@ -219,5 +233,106 @@ private bool CanAddToGroup(object parameters)
             }
             return false;
         }
+
+        private void PinToNode(object parameters)
+        {
+            
+            var nodeToPin = DynamoSelection.Instance.Selection
+                .OfType<NodeModel>()
+                .First();
+
+            MoveNoteAbovePinNode(nodeToPin);
+
+            var nodeViewModel = WorkspaceViewModel.Nodes.Where(x => x.Id == nodeToPin.GUID).FirstOrDefault();
+            nodeViewModel.RequestsSelection += NodeViewModel_RequestsSelection;
+            nodeViewModel.PropertyChanged += NodeViewModel_PropertyChanged;
+            nodeViewModel.NodeModel.PropertyChanged += NodeModel_PropertyChanged;           
+            Model.PinnedNode = nodeToPin;
+            ZIndex = Convert.ToInt32(nodeViewModel.ErrorBubble.ZIndex - 1);
+            RaisePropertyChanged(nameof(PinNode));
+        }
+
+        private void NodeModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "State")
+            {
+                MoveNoteAbovePinNode(Model.PinnedNode);
+            }
+            if (e.PropertyName == "Position")
+            {
+                MoveNoteAbovePinNode(Model.PinnedNode);
+            }
+        }
+
+        private void NodeViewModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "ZIndex")
+            {
+                var node = sender as NodeViewModel;
+                ZIndex = Convert.ToInt32(node.ErrorBubble.ZIndex - 1);
+            }
+        }
+
+        private bool CanPinToNode(object parameters)
```

Comment:
When are you expecting this to be checked? 

afaik this only gets called once, you have to manually `RaiseCanExecute` or something like that for this to be checked again, so not sure how this check makes sense? might just be me though

Reply:
I am using it in the `PinToNodeCommand` 
`pinToNodeCommand = new DelegateCommand(PinToNode, CanPinToNode);`
I thought that this will grey out the `Pin to node` option on the context menu, like `Add to Group` or `Remove from group` works, any idea on to how to make it work this way?
![image](https://user-images.githubusercontent.com/25138291/124108595-38d20100-da5e-11eb-9652-86e15fdf48e4.png)

Reply:
yea so this is where you need to use `RaiseCanExecute` every time you want to evaluate this, its the same thing as we did in the CC plugin.

Reply:
rookie mistake, thought this was automatically done with the Command binding.
Ok solved in here: 
108d77a